openedx/edx-ora2
# feat: ORA autosave enhancements

**TL;DR -** Remove "Save Response" button in favor of better autosaving functionality. Paired with better error handling and style cleanup.

JIRA: [AU-1237](https://2u-internal.atlassian.net/browse/AU-1237)

![ora-autosave](https://github.com/openedx/edx-ora2/assets/12944465/c4a52f61-8648-4a9c-ae68-86f7c0d28d1c)

**What changed?**

- Autosave improvements:
  - Increase autosave cadence to every ~4 seconds (2 second poll + 2 second wait for inactivity).
  - Update autosave text to be shorter and contain icons.
  - Update help text to indicate that responses are autosaved.
  - Remove save button and save button code.

- Error handling:
  - Update autosave logic to be able to continue after an autosave error (e.g. internet dropout).
  - Update error messaging to indicate whether the issue was frontend or backend ("Please check your internet connection" versus "Please contact support staff.")

- General UI cleanup:
  - Remove redundant submission "tip" section.
  - Add line between text and file upload sections.
  - Move "Supported file types" to its own line. Fixes a weird wrapping issue when this is too long.
  - Make padding / margins more consistent throughout.
  - Various SCSS cleanup, removing unused rules, etc.

**Example Screenshot of Team ORA with files & text:**

<img width="852" alt="Screenshot of Team ORA with files and text" src="https://github.com/openedx/edx-ora2/assets/12944465/1a0e9f12-0a92-444d-9084-7f58d327a100">

**Example screenshot of ORA with files only:**

<img width="852" alt="Screenshot of ORA with files only" src="https://github.com/openedx/edx-ora2/assets/12944465/79fca35d-f109-43e2-9a27-f5433e375a46">

---

**Developer Checklist**


**Testing Instructions**

Install locally (since this works now 😉 ) verify the following cases:
- [x] Text responses autosave at the expected cadence.
- [x] Text responses can recover and autosave after an internet drop.

Additionally, verify that the following cases still present correctly:
- [x] Text only.
- [x] Files only.
- [x] Text & files.
- [x] Team responses.
- [x] Multiprompt.

**Browser tests**

Verify that this looks correct on the following browsers:
- [x] Chrome(ium)

**Reviewer Checklist**

Collectively, these should be completed by reviewers of this PR:


FYI: @openedx/content-aurora
